src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -290,9 +290,14 @@ def visit_Lambda(
             context.body.add_array(name, shape=shape, strides=strides, dtype=dtype)
 
         # Translate the function's body
-        result: ValueExpr = self.visit(node.expr)[0]
+        result: ValueExpr | SymbolExpr = self.visit(node.expr)[0]
         # Forwarding result through a tasklet needed because empty SDFG states don't properly forward connectors
-        result = self.add_expr_tasklet([(result, "result")], "result", result.dtype, "forward")[0]
+        if isinstance(result, ValueExpr):
+            result = self.add_expr_tasklet([(result, "result")], "result", result.dtype, "forward")[
```

Comment:
We can change the tasklet arg string to "inp" so we keep the assignment on one line.
```suggestion
            result = self.add_expr_tasklet([(result, "inp")], "inp", result.dtype, "forward")[0]
```

Reply:
There, I made it even uglier :sweat_smile:  But this tasklet was really not necessary.

Reply:
Ok, I removed that part entirely. I don't think it is needed. @petiaccja can you think of a case where it is needed?

Reply:
Yeah, we talked about this before, and we removed it because it was not functionally needed. But I had to add this "forwarding" thing back because it turned out that it *is* functionally needed. When you have an empty state with input and output connectors, and you want the input connectors to be forwarded to the output connectors, you eventually have to put something in the state because otherwise it doesn't work. I just put a dummy tasklet. If you have a simpler solution that's better, but before you do anything try to add a test and don't delete the comment.

Reply:
That's fair to leave it then, the version I put there now should be adequately transformed by simplify anyways.

Reply:
Great! Unfortunately I don't remember the exact case that produced the empty state, and of course I didn't write a test for it either. We can of course use a less hacky method here as long as we don't run into problems with the source iterator IR. I'll resolve this for now.